src/niscope/system_tests/test_system_niscope.py

```diff
@@ -1,527 +1,588 @@
-    # to demonstrate that clearing the stats does in fact cause a measurable change.
-    assert uncleared_stats[0].result == uncleared_stats_2[0].result
-    assert uncleared_stats[0].stdev == uncleared_stats_2[0].stdev
-    assert uncleared_stats[0].mean == uncleared_stats_2[0].mean
-    assert uncleared_stats[0].min_val == uncleared_stats_2[0].min_val
-    assert uncleared_stats[0].max_val == uncleared_stats_2[0].max_val
-    assert uncleared_stats[0].num_in_stats == uncleared_stats_2[0].num_in_stats
-    assert uncleared_stats[0].num_in_stats != cleared_stats[0].num_in_stats
-
-
-def test_waveform_processing(multi_instrument_session):
-    test_voltage = 1.0
-    test_record_length = 1000
-    test_num_channels = 2
-    test_num_records = 3
-    multi_instrument_session.configure_vertical(test_voltage, niscope.VerticalCoupling.AC)
-    multi_instrument_session.configure_horizontal_timing(50000000, test_record_length, 50.0, test_num_records, True)
-    with multi_instrument_session.initiate():
-        multi_instrument_session.add_waveform_processing(niscope.enums.ArrayMeasurement.DERIVATIVE)
-        processed_waveforms = multi_instrument_session.channels[test_channels].fetch_measurement_stats(niscope.enums.ScalarMeasurement.MID_REF_VOLTS)
-        multi_instrument_session.clear_waveform_processing()
-        unprocessed_waveforms = multi_instrument_session.channels[test_channels].fetch_measurement_stats(niscope.enums.ScalarMeasurement.MID_REF_VOLTS)
-
-    assert len(processed_waveforms) == test_num_channels * test_num_records
-    assert len(unprocessed_waveforms) == test_num_channels * test_num_records
-    # Here the idea is to leave a large margin to not test too specifically for any returned values but to demonstrate that the waveform processing does
-    # undeniably cause a consistent shift in the values returned. The "0" exception for processed is due to the nature of derivatives -- if two samples
-    # next to each other are identical, the derivative will be 0.
-    for processed, unprocessed in zip(processed_waveforms, unprocessed_waveforms):
-        assert abs(unprocessed.result) < 1
-        assert abs(processed.result) > 1 or processed.result == 0
-
-
-def test_measurement_stats_str(multi_instrument_session):
-    test_voltage = 1.0
-    test_record_length = 1000
-    test_num_records = 1
-    multi_instrument_session.configure_vertical(test_voltage, niscope.VerticalCoupling.AC)
-    multi_instrument_session.configure_horizontal_timing(50000000, test_record_length, 50.0, test_num_records, True)
-    with multi_instrument_session.initiate():
-        measurement_stat = multi_instrument_session.channels[test_channels].fetch_measurement_stats(niscope.enums.ScalarMeasurement.NO_MEASUREMENT)
-
-    assert isinstance(measurement_stat[0].__str__(), str)
-    assert isinstance(measurement_stat[0].__repr__(), str)
-
-
-def test_get_self_cal_last_date_time(single_instrument_session):
```

Comment:
These four tests were copied as is to `TestLibrary` class. These methods aren't included in the proto file so there are four versions of these tests in `TestGrpc` where each expects a `NotImplementedError` to be raised.

Reply:
Tests for these methods are no longer present in the TestGrpc class. Since we intend to support them for gRPC very soon, I'm okay with that, so long as we have a TODO comment above them in the TestLibrary class. There is another thread where I'm requesting that comment, so I'll close this thread.